src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractJenkinsManifestMojo.java

```diff
@@ -292,6 +298,37 @@ private String getScmUrl() {
         return null;
     }
 
+    private String getSCMConnection() {
+        Scm scm = project.getScm();
+        if (scm != null) {
+            return scm.getConnection();
+        }
+        return null;
+    }
+
+    private String getGitModulePath() {
+        //execute `git rev-parse --show-toplevel` then remove that from project.baseDir and what we have left is the path
+        String topLevel = getGitRevParseOutput("--show-toplevel");
+        if (topLevel == null) {
+            return null;
+        }
+        try {
+            // rev-parse can return unix slashes on windows - so lets ensure everything is canonical
+            String f = new File(topLevel).getCanonicalPath();
+            String b = project.getBasedir().getCanonicalPath();
+            // no need to be defensive - the project must be in the git directory for git-revparse to return non null
+            String path = b.substring(f.length());
+            if (path.startsWith(File.separator)) {
+                path = path.substring(1); // for a single module project this will be the empty string
+            }
+            // normalize to Unix style
+            return path.replace(File.separatorChar, '/');
+        } catch (IOException e) {
+            getLog().warn("failed to obtain projects relative location in the Git repository", e);
```

Comment:
goes against the `fine` level if `git rev-parse HEAD` fails in the existing code - but really this should not happen - we are a git project.